Book icons

a2line opened the pull request.

Commit:
German translations: suppress superfluous phrases

Commit:
more missing german translations, mostly cousins.x.y

added 2nd de: index to "first name/first names"

added 'show all' for de,en,fr

a2line committed:
remove extra parenthesis

a2line committed:
add three new icons for books

a2line committed:
real fix for graphe centering

* remove +1 added to max_gen that broke ±1 gen. buttons
* revamp buttons: the toolbar is more vertical better fitting with fancharts
* add missing js for the hidding toolbar button

a2line committed:
make linked pages link not selectable

a2line committed:
remove html

a2line committed:
add thousand separator

a2line committed:
update missing de/it/pl translations

* remove alias entries by hand with regexp
* add missing %s to German, s/x/×/
* add a few missing cousin.x.y for French

a2line commented:
I merged and rebased #1701 in this one, translation looks good, thank you very much @ftrojahn!
![image](https://github.com/geneweb/geneweb/assets/10353895/7463994d-cd41-410c-9cda-3550760e8e77)

ftrojahn commented:
Well ... perhaps the missing-de.txt are somewhat out of date now. I think there is a tool to handle/create the
missing translations files, but don't know how to use it, so maybe I forgot to cleanup manually.

a2line commented:
I have merged what was in missing-de.txt with lex_utils tool, it's in your second commit. I have also updated the missing-de.txt file but there is a obvious issue with lexicon entries that are aliases (see https://github.com/geneweb/geneweb-contrib/issues/14)

I edit this PR manually to remove them, we don't need translations for these.

ftrojahn commented:
Ok, thanks @a2line . I'm looking forward to deliver the missing german translations soon.

a2line committed:
add uri to colab build

a2line commented:
I have added to German a few missng %s and changed “x” (times) to a “×”. German has only three real translations missing: all the `cousins.v1.v2_tt` entries are extra translations that are shown in tooltip. Those translaltions are not mandatory at all. I added them for langages like French that have multiple or old names or perhaps multiple naming systems… Note that nearly all `cousin.v1.v2` above 5.5 still have no entries. The German naming system is far simpler than French and English ones, so I added `cousin.10.1` myself!

a2line committed:
inverse sort icons

ftrojahn commented:
>  The German naming system is far simpler than French and English ones, so I added `cousin.10.1` myself!
Hello @a2line 

perhaps there has been something wrong with the lexicon mergings?
Some things were correct in my PR, changed in your's. My tooltips hold the
alternate German phrases and are now gone or wrong:

* [cousins.10.0](https://github.com/geneweb/geneweb/pull/1701/files#diff-a00298855e3e6118d197b74e90d3e29b7bb8a4d3fe75989c31bd1f21b49c92b0R5237) vs. [a2line 1704](https://github.com/geneweb/geneweb/pull/1704/files#diff-a00298855e3e6118d197b74e90d3e29b7bb8a4d3fe75989c31bd1f21b49c92b0R5392)
* [cousins.10.0 tt](https://github.com/geneweb/geneweb/pull/1701/files#diff-a00298855e3e6118d197b74e90d3e29b7bb8a4d3fe75989c31bd1f21b49c92b0R5243) vs. [a2line 1704](https://github.com/geneweb/geneweb/pull/1704/files#diff-a00298855e3e6118d197b74e90d3e29b7bb8a4d3fe75989c31bd1f21b49c92b0R5397)
* [cousins.9.0 tt](https://github.com/geneweb/geneweb/pull/1701/files#diff-a00298855e3e6118d197b74e90d3e29b7bb8a4d3fe75989c31bd1f21b49c92b0R5924) vs. [a2line 1704](https://github.com/geneweb/geneweb/pull/1704/files#diff-a00298855e3e6118d197b74e90d3e29b7bb8a4d3fe75989c31bd1f21b49c92b0R6078)
* [cousins.2.10](https://github.com/geneweb/geneweb/pull/1701/files#diff-a00298855e3e6118d197b74e90d3e29b7bb8a4d3fe75989c31bd1f21b49c92b0R5325) vs [a2line 1704](https://github.com/geneweb/geneweb/pull/1704/files#diff-a00298855e3e6118d197b74e90d3e29b7bb8a4d3fe75989c31bd1f21b49c92b0R5479)

(just checked some). Could it be, that you merged my missing-de.txt, too? Then, that may be the culprit. It was only my "scratchpad" and got accidentally not cleaned up. I'll have a look soon and maybe make another PR.
I'll have to recheck this in the "lexcheck=1" view (great setting, btw. ;) )

A bit OT: I find it a bit strange, that the (lex_util?) sorting does it literally instead of numerically, e.g. 10.10 after 10.1 - instead of 10.10 after 10.9 etc. Is this fixable?

Since the German phrases are somewhat predictable, question: should I provide all the "higher numbers" like 6.12 ff., 5.14 ff., 4.11 etc. or could there be some programmatically generation?

hgouraud commented:
> A bit OT: I find it a bit strange, that the (lex_util?) sorting does it literally instead of numerically, e.g. 10.10 after 10.1 - instead of 10.10 after 10.9 etc. Is this fixable?

possibly, but this would require significant effort which is has lower priority !!

ftrojahn commented:
> possibly, but this would require significant effort which is has lower priority !!

Thanx @hgouraud  for your fast reply, never mind, it was just an idea. Of course there are things with higher prio.

a2line commented:
10.0 is here just moved above by the sorting process.
9/10.0_tt have been lost (I was wondering why they where the same as non_tt entries !)
2.10 is the strangest one because it got a totally wrong translation that is not yours, was also sorted… it's probably the tool that messed up doing merge + sort…

I push force to fix up your commit, here is diff. for proof of what has been added.
![image](https://github.com/geneweb/geneweb/assets/10353895/f8fc808d-5db9-4df0-95e6-8da584ca0a11)

> Since the German phrases are somewhat predictable, question: should I provide all the "higher numbers" like 6.12 ff., 5.14 ff., 4.11 etc. or could there be some programmatically generation?

It could be great if all human could have the same notation and that it would be readable, smartly generated and meaningfull! :D

a2line committed:
add missing url pathname

give only one URL in cgi, use _f/_w instead of w= parameter

hgouraud committed:
enforce unicity of env variables
